### DISCUSSION TITLE
Lesson 2: Storage Factory, variable simplestorage not initialised

### DISCUSSION THREAD

**Discussion by manavnotnani** [smartcontractkit/full-blockchain-solidity-course-py#471] [Q&A] [answered]
In function createSimpleStorageContract(), the compiler is telling that simplestorage is not initialized above. But I can see it is running in the course. idk why my code is showing error?
![Screenshot 2021-12-02 110842](https://user-images.githubusercontent.com/76724460/144364282-7cc5de40-a142-445f-948b-f4fcdb75bc6a.png)

**Comment by cromewar** [comment, marked as answer]
hello @manavnotnani thats strange, make sure you don't have a typo on  `simpleStorageArray.push(simpleStorage);`, anyways could you share your code in order we could review it please?

**Reply by manavnotnani** [reply]
I checked that thrice. It didn't have any typo error. Please tell whether we have to initialize a  variable before using it in function or it can be done in function itself( I'm taking about public scope variable)

**Reply by cromewar** [reply]
You should initialize the simple storage array before indeed.

**Reply by widmung89** [reply]
I seem to get that same error as well when I do:

```Solidity
import "./SimpleStorage.sol";

contract StorageFactory {
    function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
    }

}

```
How do I resolve the error message for Unused local variable?

**Reply by cromewar** [reply]
Well you should just use that variable, if you continue with the course you'll see that the function will be:

```Solidity
   function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
        simpleStorageArray.push(simpleStorage);
    }
```

So as you are using the variable right after declaring it, the warning will disappear.
